File: 简单工厂模式/基本实现/Api_Normal.java
interface Api {
	public void test(String s);
}

class Impl implements Api {
	public void test(String s ){
		System.out.println(s);
	}
}

public class Api_Normal {
	public static void main(String[] args){
		Api api = new Impl();
		api.test("test");
	}
}

File: 简单工厂模式/基本实现/Api_Normal.php
<?php 

interface Api {
	public function test($string);
}


class Impl implements Api {
	public function test($string=""){
		echo $string;
	}
}

function work(Api $api,$str){
	$api->test($str);
}

$api = new Impl();
work($api,"world");